packages/metro/src/ModuleGraph/worker/__tests__/collectDependencies-test.js

```diff
@@ -42,45 +43,206 @@ const opts = {
   dependencyMapName: null,
 };
 
-it(`collects require context arguments`, () => {
-  const ast = astFromCode(`
-  const a = require.context('./', false, /foobar/m, 'eager');
-  const b = require.context('./baz');
-`);
-  const {dependencies, dependencyMapName} = collectDependencies(ast, {
-    ...opts,
-    unstable_allowRequireContext: true,
+describe(DefaultModuleDependencyRegistry, () => {
```

Comment:
```suggestion
describe('DefaultModuleDependencyRegistry', () => {
```

Reply:
I see the appeal of testing the dependency registry in isolation, but I would prefer that we write the tests in terms of `collectDependencies`. I'd go so far as to say that the entire concept of a pluggable dependency registry may be redundant, and I'll look into killing it after this PR lands. Selfishly, I'd appreciate having tests that I don't have to rewrite when I do that :D

Reply:
I can write tests at both levels. Feel free to delete the lowers level ones later. 

I exposed/tested this function in particular since I can see people extending the collector and wanting to reuse the key resolver.